process/smartContract/hooks/blockChainHook.go

```diff
@@ -475,7 +456,7 @@ func (bh *BlockChainHookImpl) IsPayable(sndAddress []byte, recvAddress []byte) (
 // FilterCodeMetadataForUpgrade will filter the provided input bytes as a correctly constructed vmcommon.CodeMetadata bytes
 // taking into account the activation flags for the future flags. This should be used in the upgrade SC process
 func (bh *BlockChainHookImpl) FilterCodeMetadataForUpgrade(input []byte) ([]byte, error) {
-	if !bh.flagFilterCodeMetadataEnableEpoch.IsSet() {
+	if !bh.enableEpochsHandler.IsPayableBySCFlagEnabled() {
```

Comment:
Since this change will obscure the old code, I suggest to add something like:
```GO
filterCodeMetadataEnableEpoch := bh.enableEpochsHandler.IsPayableBySCFlagEnabled()
if !filterCodeMetadataEnableEpoch{
////......

```